megamek/src/megamek/client/ui/swing/MegaMekGUI.java

```diff
@@ -536,7 +543,24 @@ public String getDescription() {
             // I want a file, y'know!
             return;
         }
-        HostDialog hd = new HostDialog(frame);
+
+        IGame newGame = null;
+        try (InputStream is = new FileInputStream(fc.getSelectedFile()); InputStream gzi = new GZIPInputStream(is)) {
+            XStream xstream = SerializationHelper.getXStream();
+            newGame = (IGame) xstream.fromXML(gzi);
+        } catch (Exception e) {
+            MegaMek.getLogger().error("Unable to load file: " + fc.getSelectedFile(), e);
+        }
```

Comment:
Currently when a game doesn't load, it continues to the host dialog (and notices later, that it cant load). Maybe add 

JOptionPane.showMessageDialog(frame, Messages.getString("MegaMek.LoadGameAlert.message"),
                    Messages.getString("MegaMek.LoadGameAlert.title"), JOptionPane.ERROR_MESSAGE);
return;

to the catch block?

Reply:
Yeah, that's a fair point.  This briefly crossed my mind.  I chose to not deal with it because I figured the old path would be fine (and it was easier to do nothing about it).  But it does make sense to handle things earlier, before the players enter additional information.  I had assumed that this loading isn't likely to fail, but I guess it is if you load saved games across versions,

Reply:
I don't think we have any expectation of support for megamek games being loaded across versions. That being said, some error handling won't go amiss.

Reply:
That's exactly what I meant: we don't have any expectation of supporting games across versions and those are the most likely case of loading errors.